examples/threading/threading.c

```diff
@@ -2,18 +2,82 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <pthread.h>
+#include <errno.h>
 
 // Optional: use these functions to add debug or error prints to your application
 #define DEBUG_LOG(msg,...)
 //#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
-#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)  //variable argument
 
 void* threadfunc(void* thread_param)
 {
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    int ret;
+    int errnum;
+    
+    
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    //printf("thread_param: %p\n\r", thread_func_args);
+    //printf("mutex_info: %p\n\r", thread_func_args->mutex_info);
+    
+    printf("Thread start. Going to sleep for %d\n\r", thread_func_args->wait_obtain_mutex);
+    
+    ret = usleep((thread_func_args->wait_obtain_mutex)*1000);
+    if(ret != 0)
+    {
+    	//error
+    	errnum = errno;
+    	/* mutex_lock() call would technically be blocked until the mutex is available. 
+    	So this would always be zero but still no harm in checking it. 
+    	In case there are other bugs.*/
+    	printf("Error: %d \n\r", errnum);
+    }
+    
+    printf("Obtaining mutex. Going to sleep for %d \n\r", thread_func_args->wait_release_mutex);
+    
+    ret = pthread_mutex_lock(thread_func_args->mutex_info);
+    printf("Obtained mutex\n\r");
+    if(ret != 0)
+    {
+    	//error
+    	errnum = errno;
+    	/* mutex_lock() call would technically be blocked until the mutex is available. 
+    	So this would always be zero but still no harm in checking it. 
+    	In case there are other bugs.*/
+    	printf("Error: %d \n\r", errnum);
+    }
+    
+    printf("Going to sleep\n\r");
+    //wait()
+    ret = usleep((thread_func_args->wait_release_mutex)*1000);
+    if(ret != 0)
+    {
+    	//error
+    	errnum = errno;
+    	/* mutex_lock() call would technically be blocked until the mutex is available. 
+    	So this would always be zero but still no harm in checking it. 
+    	In case there are other bugs.*/
+    	printf("Error: %d \n\r", errnum);
+    }
+    printf("Unlocking mutex\n\r");
+    
+    ret = pthread_mutex_unlock(thread_func_args->mutex_info);
+    if(ret != 0)
+    {
+    	//error
+    	/* mutex_unlock() call would immediately release the mutex and return zero.
+    	Wouldn't harm to still check*/
+    	printf("Error: %d \n\r", errnum);
+    }
+    printf("End thread\n\r");
+    
+    thread_func_args->thread_complete_success = true;
```

Comment:
@Anshul-99 do you think just printing of errors here is sufficient? what will happen in case of mutex lock/unlock 
 or usleep failure ? you're still returning true here!